cranelift/object/src/backend.rs

```diff
@@ -551,6 +551,24 @@ impl ObjectModule {
                     symbol
                 }
             }
+            ir::ExternalName::TlsIndex => {
+                let name = b"_tls_index";
```

Comment:
Maybe have a variant similar to LibCall and add a function similar to `self.libcall_names` to get the symbol name?

Reply:
I've added a `KnownSymbol` struct. However, if we had a lookup function, wouldn't that lead to all kinds of noncompliance with the COFF/PE format?

Reply:
The lookup function would default to the correct name, but there may be cases where you want a different name. For example I plan to support tls in cg_clif's jit mode by using the TlsGd tls model and then redirecting `__tls_get_addr` to a function implemented in cg_clif which does the tls emulation, as libc doesn't have knowledge of the tls data objects used by the jitted code.